examples/platform/beken/common/CommonDeviceCallbacks.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/util/af-types.h>
+#include <common/CHIPDeviceManager.h>
+#include <platform/CHIPDeviceLayer.h>
+
+class CommonDeviceCallbacks : public chip::DeviceManager::CHIPDeviceManagerCallbacks
+{
+public:
+    virtual void DeviceEventCallback(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg);
+
+private:
+    void OnInternetConnectivityChange(const chip::DeviceLayer::ChipDeviceEvent * event);
+};
+
+class DeviceCallbacksDelegate
+{
+public:
+    static DeviceCallbacksDelegate & Instance()
+    {
+        static DeviceCallbacksDelegate instance;
+        return instance;
+    }
+    virtual void OnIPv4ConnectivityEstablished() {}
+    virtual void OnIPv4ConnectivityLost() {}
+    DeviceCallbacksDelegate * mDelegate = nullptr;
```

Comment:
Class members should not be public. Mark private since you have a setter and a getter.

Reply:
OK. We will fix it.